driver.py

```diff
@@ -0,0 +1,201 @@
+from abc import ABC, abstractmethod
+from processors.drop_close import DropClose
+from processors.outlier_removal import GaussOutlierRemoval
+from processors.distance import Distance
+from processors.step_range import StepRange
+from processors.processor import Processor
+from data_loaders.ecomapper_loader import EcomapperLoader
+from data_loaders.data_loader import DataLoader  
+from data_loaders.json_loader import JsonLoader
+from models.gpr import GPR
+from models.svr import SV
+from models.model_manager import ModelManager
+from models.model import Model
+from visualizers.visualization import Visualization
+from visualizers.two_d_vis import TwoDVis
+from visualizers.three_d_vis import ThreeDVis
+import matplotlib as mpl 
+from matplotlib import pyplot as plt
+import argparse
+
+
+
+parser = argparse.ArgumentParser()
```

Comment:
Even if this is a script that's meant to be run directly, can you collect all the code below into separate functions? For inspiration, something like this:
```
def parse_args():
    # build and run the argparser here

def main():
    ...

if __name__ == '__main__':
    main()
```

Reply:
Wait so what is the main() method actually doing? Because all of what I'm doing from lines 23 - 40 is adding arguments, which I would have to do anyway in the parse_args() method you wrote above, so is the main() method supposed to execute the action of changing the value of each variable, if it is manually changed by the user in the terminal using the argparser?

Reply:
It's mostly a style issue, but main() should be the top-level function that calls into everything else - there shouldn't be any other logic executed in the global scope. So main() should first build the argparser, then read the user-specified args, then preprocess the data, then run the GP, etc...

It's up to you if you want to stuff everything in one large function, but good coding style would dictate writing separate functions for each subtask.

Reply:
Ok I divided it into methods, but I have to keep returning massive tuples because almost all the variables are needed in every method